src/common/layout/SortDropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import * as S from './styles';
+import usePopover from '@hooks/usePopover';
+import { objectKeys } from '@modern-kit/utils';
+
+/**
+ * @template T
+ * @param {T} options - 드롭다운에 표시될 옵션 객체. 키는 라벨로 사용되며 값은 관련 데이터를 나타냅니다.
+ * @param {keyof T} selectedOption - 현재 선택된 옵션.
+ * @param {(option: keyof T) => void} onSelectOption - 옵션 선택 시 호출되는 콜백 함수.
+ */
+interface SortDropDownProps<T> {
+  options: T;
+  selectedOption: keyof T;
+  onSelectOption: (option: keyof T) => void;
+}
```

Comment:
```suggestion
interface SortDropDownProps<T extends Record<PropertyKey, any>> {
  options: T;
  selectedOption: keyof T;
  onSelectOption: (option: keyof T) => void;
}
```
T가 객체라면 `T extends Record<PropertyKey, any>` 로 타입을 제한해주세요

Reply:
@rhehfl 님께서 언급해주신 부분과 비슷한 것 같습니다.

확실히 이렇게 하면 T를 강제 객체 타입으로 제한을 둘 수 있겠군요😅
아무래도 현재 객체 형식으로 받는 것보다 배열 형태로 받는 게 더 좋을 것 같다는 의견이 있어서 타입이 다르게 정의 될 것 같긴 하지만 위 부분을 참고하도록 하겠습니다!
 
항상 말씀해주시는 타입을 좁히는 행위가 참 중요한 것 같네요 ㅎㅎ 리뷰 감사합니다!